### PR TITLE
Refactor:Use pointers instead of function calls in OpenMP

### PR DESCRIPTION
**Pull request by A-006**
1.There are many reusable functions in the Gint OpenMP code, and pointers can be used instead to make the code more standardized.
2. The results of the OpenMP version of dvlocal are not saved in pvpdrx. Record and save them.

### CODE REVIEW

File: source/module_hamilt_lcao/module_gint/gint_force_cpu_interface.cpp [hunk 1]
```diff
@@ -39,47 +40,110 @@ void Gint::cpu_force_interface(Gint_inout* inout) {
...
+                                        block_size, cal_flag,dpsir_ylm_x.get_ptr_2D(), dpsir_ylm_y.get_ptr_2D(),
+                                        dpsir_ylm_z.get_ptr_2D(),dpsirr_ylm.get_ptr_2D());
+
+            //do integration to get stress
+            this-> cal_meshball_stress(na_grid, block_index, psir_vlbr3_DM.get_ptr_1D(), 
+                                        dpsirr_ylm.get_ptr_1D(), svl_dphi_thread);
+        }
+
+    //release memories
+        delete[] block_iw;
```
Comment (mohanchen) [inline source/module_hamilt_lcao/module_gint/gint_force_cpu_interface.cpp]: stop using new and delete for these variables

File: source/module_hamilt_lcao/module_gint/gint_force_cpu_interface.cpp [hunk 2]
```diff
@@ -127,44 +191,166 @@ void Gint::cpu_force_meta_interface(Gint_inout* inout) {
...
+            //partial z of vtau part
+            Gint_Tools::cal_dpsirr_ylm(*this->gridt, this->bxyz, na_grid, grid_index, block_index, block_size, cal_flag,
+                ddpsir_ylm_xz.get_ptr_2D(), ddpsir_ylm_yz.get_ptr_2D(), ddpsir_ylm_zz.get_ptr_2D(), array.get_ptr_2D());
+            //do integration to get stress
+            this-> cal_meshball_stress(na_grid, block_index, dpsirz_v_DM.get_ptr_1D(),
+                array.get_ptr_1D(), svl_dphi_thread);
+        }
+
+    //release memories
+        delete[] block_iw;
```
Comment (mohanchen) [inline source/module_hamilt_lcao/module_gint/gint_force_cpu_interface.cpp]: stop using new and delete

File: source/module_hamilt_lcao/module_gint/gint_force_cpu_interface.cpp (hunk 1 above)
Comment (A-006) [reply source/module_hamilt_lcao/module_gint/gint_force_cpu_interface.cpp]: The modifications have been completed, including vindex, vldr3, block_iw, block_size, block_index. Meanwhile, cal_flag has been refactored using Array_Pool, and comments in Doxygen format have been added.

File: source/module_hamilt_lcao/module_gint/gint_force_cpu_interface.cpp (hunk 2 above)
Comment (A-006) [reply source/module_hamilt_lcao/module_gint/gint_force_cpu_interface.cpp]: The modifications have been completed, including vindex, vldr3, block_iw, block_size, block_index. Meanwhile, cal_flag has been refactored using Array_Pool, and comments in Doxygen format have been added.

Comment (A-006) [reply source/module_hamilt_lcao/module_gint/gint_force_cpu_interface.cpp]: The modifications have been completed through vector, including vindex, vldr3, block_iw, block_size, block_index. Meanwhile, cal_flag has been refactored using Array_Pool, and comments in Doxygen format have been added.

File: source/module_hamilt_lcao/module_gint/gint_rho_cpu_interface.cpp [hunk 3]
```diff
@@ -53,31 +122,136 @@ void Gint::cpu_tau_interface(Gint_inout* inout) {
...
+                    vindex.data(),
+                    dpsir_ylm_x.get_ptr_2D(), dpsir_ylm_y.get_ptr_2D(), dpsir_ylm_z.get_ptr_2D(),
+                    dpsix_DM.get_ptr_2D(), dpsiy_DM.get_ptr_2D(), dpsiz_DM.get_ptr_2D(),
+                    inout->rho[is]);
+            }
+        }
     }
+#ifdef _OPENMP
+}
+#endif
```
Comment (dzzz2001) [inline source/module_hamilt_lcao/module_gint/gint_rho_cpu_interface.cpp]: This piece of code looks odd, I believe there is a better way to refactor it.

Comment (A-006) [reply source/module_hamilt_lcao/module_gint/gint_rho_cpu_interface.cpp]: About which part? OpenMP or Gint's code?

Comment (A-006) [reply source/module_hamilt_lcao/module_gint/gint_rho_cpu_interface.cpp]: The issue of removing the OpenMP judgment condition has been resolved.

**Review by mohanchen — APPROVED** [review]

**Pull request merged**